src/main/java/com/box/sdk/BoxSignRequest.java

```diff
@@ -0,0 +1,812 @@
+             */
+            public List<BoxFile.Info> getFiles() {
+                return this.files;
+            }
+
+            /**
+             * Gets the flag indicating whether a change to the document is processing and the PDF may be out of date.
+             * It is recommended to wait until processing has finished before downloading the PDF.
+             * Webhooks are not sent until processing has been completed.
+             *
+             * @return true if files are ready to download, otherwise false.
+             */
+            public boolean getIsReadyToDownload() {
+                return this.isReadyToDownload;
+            }
+        }
+    }
+
+    /**
+     * Represents a status of the sign request.
+     */
+    public enum BoxSignRequestStatus {
+
+        /**
+         * Converting status.
+         */
+        Converting("converting"),
+
+        /**
+         * Created status.
+         */
+        Created("created"),
+
+        /**
+         * Sent status.
+         */
+        Sent("status"),
+
+        /**
+         * Viewed status.
+         */
+        Viewed("viewed"),
+
+        /**
+         * Signed status.
+         */
+        Signed("signed"),
+
+        /**
+         * Cancelled status.
+         */
+        Cancelled("cancelled"),
+
+        /**
+         * Declined status.
+         */
+        Declined("declined"),
+
+        /**
+         * Error converting status.
+         */
+        ErrorConverting("error_converting"),
+
+        /**
+         * Error sending status.
+         */
+        ErrorSending("error_sending"),
+
+        /**
+         * Expired status.
+         */
+        Expired("expired");
+
+        private final String jsonValue;
+
+        private BoxSignRequestStatus(String jsonValue) {
+            this.jsonValue = jsonValue;
+        }
+
+        static BoxSignRequestStatus fromJSONString(String jsonValue) {
+            if (jsonValue.equals("converting")) {
+                return Converting;
+            } else if (jsonValue.equals("created")) {
+                return Created;
+            } else if (jsonValue.equals("sent")) {
+                return Sent;
+            } else if (jsonValue.equals("viewed")) {
+                return Viewed;
+            } else if (jsonValue.equals("signed")) {
+                return Signed;
+            } else if (jsonValue.equals("cancelled")) {
+                return Cancelled;
+            } else if (jsonValue.equals("declined")) {
+                return Declined;
+            } else if (jsonValue.equals("error_converting")) {
+                return ErrorConverting;
+            } else if (jsonValue.equals("error_sending")) {
+                return ErrorSending;
+            } else if (jsonValue.equals("expired")) {
+                return Expired;
+            } else {
+                throw new IllegalArgumentException("The provided JSON value isn't a valid sing request status.");
+            }
```

Comment:
This is a good candidate for using a switch statement which should allow better performance

Reply:
Strings in switch are not supported in -source 1.6.